src/main/resources/+ciplugins/+jenkins/BuildReportPlugin.m

```diff
@@ -6,20 +6,24 @@
 
         function runTaskGraph(plugin, pluginData)
             runTaskGraph@matlab.buildtool.plugins.BuildRunnerPlugin(plugin, pluginData);
-            fID = fopen(fullfile(getenv("WORKSPACE"),'.matlab/buildArtifact.json'), 'w');
-            taskDetails = struct();
-            for idx = 1:numel(pluginData.TaskResults)
-                taskDetails(idx).name = pluginData.TaskResults(idx).Name;
-                taskDetails(idx).description = pluginData.TaskGraph.Tasks(idx).Description;
-                taskDetails(idx).failed = pluginData.TaskResults(idx).Failed;
-                taskDetails(idx).skipped = pluginData.TaskResults(idx).Skipped;
-                taskDetails(idx).duration = string(pluginData.TaskResults(idx).Duration);
+            [fID, msg] = fopen(fullfile(getenv("WORKSPACE"),'.matlab/buildArtifact.json'), 'w');
+
+            if fID == -1
+                warning('BuildTool:artifactFileWarnning','Could not open a file for Jenkins build result table due to: %s', msg);
+            else
+                taskDetails = struct();
+                for idx = 1:numel(pluginData.TaskResults)
+                    taskDetails(idx).name = pluginData.TaskResults(idx).Name;
+                    taskDetails(idx).description = pluginData.TaskGraph.Tasks(idx).Description;
+                    taskDetails(idx).failed = pluginData.TaskResults(idx).Failed;
+                    taskDetails(idx).skipped = pluginData.TaskResults(idx).Skipped;
+                    taskDetails(idx).duration = string(pluginData.TaskResults(idx).Duration);
+                end
+                a = struct("taskDetails",taskDetails);
+                s = jsonencode(a,"PrettyPrint",true);
+                fprintf(fID, '%s',s);
+                fclose(fID);
```

Comment:
This line will also error if `fopen` fails because it will run `fclose(-1)`

Reply:
this is else block, so i don't think it would fail I tried that already by explicitly passing -1 to fID after if block it exits.